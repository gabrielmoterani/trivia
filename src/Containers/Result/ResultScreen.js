import React from 'react';
import { connect } from 'react-redux';
import { Layout, Text, Button, Icon } from 'react-native-ui-kitten';
import { List, ListItem } from 'react-native-ui-kitten';
import BaseLayout from '../../Components/BaseLayout';
import GameActions from '../../Stores/Game/Actions';
import GameSelectors from '../../Stores/Game/Selectors';
import i18n from '../../Locales/i18n';
import Style from './ResultScreenStyle';

function ResultScreen(props) {
  const renderAnswer = ({ item }) => {
    return (
      <ListItem
        title={item.question.question}
        description={item.question.category}
        icon={() => <Icon name={item.correct ? 'checkmark-circle-2-outline' : 'close-outline'} />}
      />
    );
  };

  return (
    <BaseLayout>
      <Layout style={Style.score}>
        <Text category="h6">{i18n.t('result.score')}: {props.score}/{props.answers.length}</Text>
      </Layout>
      <Layout style={Style.answers}>
        <List data={props.answers} renderItem={renderAnswer} />
      </Layout>
      <Button onPress={props.resetGame}>
        {i18n.t('result.backToMain')}
      </Button>
    </BaseLayout>
  );
}

const mapStateToProps = (state) => ({
  answers: GameSelectors.answers(state),
  score: GameSelectors.score(state),
});

const mapDispatchToProps = (dispatch) => ({
  resetGame: () => dispatch(GameActions.resetGame()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ResultScreen);
