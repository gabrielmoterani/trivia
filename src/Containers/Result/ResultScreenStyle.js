import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    flex: 0.1,
    justifyContent: 'flex-start',
    marginTop: 50,
  },
  main: {
    flex: 0.8,
    alignItems: 'center',
    justifyContent: 'center',
  },
  score: {
    alignItems: 'center',
  },
  answers: {
    width: 325,
    height: 'auto',
    maxHeight: 400,
    marginVertical: 25,
  },
  footer: {
    flex: 0.1,
    justifyContent: 'flex-end',
    marginBottom: 50,
  },
});
