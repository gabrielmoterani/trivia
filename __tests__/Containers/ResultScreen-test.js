import React from 'react';
import { shallow } from 'enzyme';
import { ResultScreen } from '../../src/Containers/Result/ResultScreen';

const initialProps = {
  answers: [],
  score: 0,
  resetGame: jest.fn(),
};

const setup = (props = initialProps) => shallow(<ResultScreen {...props} />);

describe('Test ResultScreen container', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a start over button', () => {
    const startOverBtn = wrapper.find('ButtonComponent');
    expect(startOverBtn).toExist();
  });

  it('should properly trigger action when click start over', () => {
    const startOverBtn = wrapper.find('ButtonComponent').first();
    startOverBtn.props().onPress();

    expect(initialProps.resetGame).toHaveBeenCalled();
  });
});
