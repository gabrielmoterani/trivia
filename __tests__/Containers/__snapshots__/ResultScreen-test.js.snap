// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ResultScreen container should render properly 1`] = `
<BaseLayout>
  <LayoutComponent
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TextComponent
      category="h6"
    >
      Your Score
      : 
      0
      /
      0
    </TextComponent>
  </LayoutComponent>
  <LayoutComponent
    style={
      Object {
        "flex": 1,
        "marginVertical": 25,
        "width": "90%",
      }
    }
  >
    <ListComponent
      data={Array []}
      renderItem={[Function]}
    />
  </LayoutComponent>
  <ButtonComponent
    onPress={[MockFunction]}
  >
    Back to main screen
  </ButtonComponent>
</BaseLayout>
`;
